content/angular/pt/data.md

```diff
@@ -0,0 +1,306 @@
+---
+title: "Ligação de dados"
+tocTitle: "Dados"
+description: "Aprendizagem da metodologia de ligação de dados ao componente interface utilizador"
+---
+
+## Ligação de dados
+
+Até agora foram criados componentes sem estado e isolados, o que é fantástico para Storybook, mas em última análise não são úteis até que for fornecido algum tipo de dados da aplicação
+
+Este tutorial não foca particularmente na construção de uma aplicação, como tal não vamos aprofundar muito este aspeto. Mas será feito um aparte para olhar para um padrão comum para ligação de dados com componentes contentor.
+
+## Componentes contentor
+
+O componente `TaskListComponent` na sua presente forma é um componente de "apresentação" (ver [este post no blog](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0)), de forma que este não comunica com nada externo além de si.
+Para conter dados, irá ser necessário um "contentor".
+
+Este exemplo utiliza [ngxs](https://ngxs.gitbook.io/ngxs/), que é a biblioteca que adopta os principios Redux/ngrx, mas foca-se na redução de código denominado padão e oferece uma forma considerada mais _angular-y_ para a gestão do estado, de forma a construir um modelo de dados simples para a aplicação. No entanto o padrão que irá ser usado, pode ser aplicado a outras bibliotecas de gestão de dados tal como [ngrx/store](https://github.com/ngrx/platform).
+
+Irá começar por uma ligeira refactorização da aplicação, será necessário criar duas novas pastas: `containers` e `components` respectivamente. Após isto, irão ser movidos os ficheiros `task.component.ts` and `task-list.component.ts`, (assim como os ficheiros de estórias `.stories.ts` correspondentes) para dentro desta última. Com isto, será necessário atualizar todas e quaisquer referências enumeradas pela cláusula import.
+
+Em seguida irá ser construido um contentor de estados simples que irá responder de acordo a qualquer ação que altere o estado das tarefas, isto num ficheiro denominado `src/tasks/state/task.state.ts` (que é mantido de forma simples intencionalmente):
+
+```typescript
+import { State, Selector, Action, StateContext } from '@ngxs/store';
+import { Task } from '../task.model';
+
+export const actions = {
+  ARCHIVE_TASK: 'ARCHIVE_TASK',
+  PIN_TASK: 'PIN_TASK',
+};
+
+export class ArchiveTask {
+  static readonly type = actions.ARCHIVE_TASK;
+
+  constructor(public payload: string) {}
+}
+
+export class PinTask {
+  static readonly type = actions.PIN_TASK;
+
+  constructor(public payload: string) {}
+}
+
+// The initial state of our store when the app loads.
```

Comment:
Could you translate these comments into the code as well?? 🙏🏻

Reply:
Going to reply just here, to avoid repetition, both code comments translations are done. Just going to commit the change in a bit.

Reply:
great